packages/graph-editor/src/components/flow/nodes/sets/ExternalSetNode.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import { IconButton, Spinner, Stack, Text } from '@tokens-studio/ui';
+import {
+    ClipboardCopyIcon,
+    DownloadIcon,
+} from '@radix-ui/react-icons';
+import { Handle, HandleContainer } from '../../handles.tsx';
+import { flatTokensRestoreToMap } from '#/utils/index.ts';
+import { NodeProps } from 'reactflow';
+import {
+    EXTERNAL_SET_ID,
+    node,
+} from '@tokens-studio/graph-engine/nodes/set/externalTokens.js';
+import { WrapNode, useNode } from '../../wrapper/nodeV2.tsx';
+import copy from 'copy-to-clipboard';
+import { TokenSetHandles } from './tokensHandles.tsx';
+import { useExternalData } from '#/context/ExternalDataContext.tsx';
+
+
+type ExternalSetData = {
+    urn: string
+    name: string;
+};
+
+const ExternalSetNode: FC<NodeProps<ExternalSetData>> = () => {
+    const { setControls, setTitle, state, output, setState } = useNode();
```

Comment:
We need to expose the ephemeral property through useNode and the rest of the ephemeral lifecycle. That would remove the complexity of each node managing its lifecycle to load external data